### ISSUE TITLE
[clang-tidy-19] readability-container-size-empty: false positive when used in value template parameters

### ISSUE THREAD

**Issue by alexkaratarakis** [llvm/llvm-project#118418]
Min Repro:

`test.cpp`:
```C++
#include <array>

template <auto LHS, auto RHS>
inline constexpr bool ARE_EQUAL = (LHS == RHS);

static_assert(ARE_EQUAL<5, std::array<int, 5>{}.size()>); // Diagnostic triggers here

static_assert(5 == std::array<int, 5>{}.size()); // Diagnostic does not trigger here
```

`.clang-tidy`:
```
Checks: >
  readability-container-size-empty,
```

And running this through clang-tidy:
```
$ clang-tidy-19 test.cpp 
2 warnings generated.
/temp/test.cpp:6:28: warning: the 'empty' method should be used to check for emptiness instead of 'size' [readability-container-size-empty]
    6 | static_assert(ARE_EQUAL<5, std::array<int, 5>{}.size()>);
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                            !std::array<int, 5>{}.empty()
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/array:195:7: note: method 'array'::empty() defined here
  195 |       empty() const noexcept { return size() == 0; }
      |       ^

```

Note that this did not trigger with `clang-tidy-18`.
